rakam-postgresql/src/main/java/org/rakam/postgresql/analysis/FastGenericFunnelQueryExecutor.java

```diff
@@ -133,10 +145,11 @@ public QueryExecution query(String project, List<FunnelStep> steps, Optional<Str
                     List<List<Object>> newResult = new ArrayList<>();
                     List<SchemaField> metadata;
 
-                    if (dimension.isPresent()) {
+                    if (dimension.isPresent() || segment.isPresent()) {
+                        String schemaFieldName = dimension.isPresent() ? "dimension" : "segment";
                         metadata = ImmutableList.of(
                                 new SchemaField("step", STRING),
-                                new SchemaField("dimension", STRING),
+                                new SchemaField(schemaFieldName, STRING),
```

Comment:
No need to rename it to `segment` since it's also part of the dimension.

Reply:
Reverted that